lib/rex/logging/log_dispatcher.rb

```diff
@@ -176,5 +198,23 @@ def get_log_level(src)
   $dispatcher.get_level(src)
 end
 
+def get_global_log_level(src)
+  level = get_log_level(src)
+
+  # If the source has no associated log_level, the default log level is used
+  unless level
+    level = LEV_3
+  end
+
+  level
+end
+
+def build_error_details(error, global_log_level)
```

Comment:
This looks like it will introduce a global function, I don't think that's something we'd want 👀 

Would it be easier to just inline the usage?

Reply:
Missed that this would be in the Global namespace. It'd be easier but smelly due to the duplication of logic